src/safir/metadata.py

```diff
@@ -63,58 +67,23 @@ def setup_metadata(
         Used as the ``documentation_url`` metadata.
     project_urls, Source code
         Used as the ``respository_url``.
-
-    The configuration object (``app["safir/config"]``) also provides metadata:
-
-    config.name
-        Used as the ``name`` metadata field.
-
-    **Metadata schema**
-
-    The metadata is stored as a `dict` in the ``"safir/metadata"`` key of the
-    application:
-
-    .. code-block:: json
-
-       {
-         "name": "safirdemo",
-         "version": "0.1.0",
-         "description": "Demonstration of a Safir-based app.",
-         "repository_url": "https://github.com/lsst-sqre/safirdemo",
-         "documentation_url": "https://github.com/lsst-sqre/safirdemo"
-       }
     """
     pkg_metadata: Message = metadata(package_name)
-
-    try:
-        config = app["safir/config"]
-    except KeyError:
-        raise RuntimeError(
-            "Application does not have a 'safir/config' key. "
-            "Add configuration to the application before running "
-            "setup_metadata."
-        )
-
-    meta: Dict[str, Any] = {
-        # Use configured name in case it is dynamically changed.
-        "name": config.name,
-        # Get metadata from the package configuration
-        "version": pkg_metadata.get("Version", "0.0.0"),
-        "description": pkg_metadata.get("Summary", None),
-        "repository_url": get_project_url(pkg_metadata, "Source code"),
-        "documentation_url": pkg_metadata.get("Home-page", None),
-    }
-    meta.update(**kwargs)
-    app["safir/metadata"] = meta
+    return Metadata(
+        name=application_name,
+        version=pkg_metadata.get("Version", "0.0.0"),
+        description=pkg_metadata.get("Summary", None),
+        repository_url=get_project_url(pkg_metadata, "Source code"),
+        documentation_url=pkg_metadata.get("Home-page", None),
+    )
 
 
 def get_project_url(meta: Message, label: str) -> Optional[str]:
-    """Get a specific URL from the ``project_urls`` key of a package's
-    metadata.
+    """Get a specific URL from a package's ``project_urls`` metadata.
 
     Parameters
     ----------
-    meta : `email.message.Message`
+    meta : ``email.message.Message``
```

Comment:
Did the intersphinx link not work? it should resolve to https://docs.python.org/3/library/email.compat32-message.html?highlight=email#email.message.Message

Reply:
Oh, it did work. I thought it wasn't working, but it's complaining about the type signature instead. (Sphinx isn't quite up to parsing type signatures properly.) Will fix.